### DISCUSSION TITLE
Why does readOnly = true means you can't clear the date?

### DISCUSSION THREAD

**Discussion by FerMPY** [onesine/react-tailwindcss-datepicker#112] [Q&A] [answered]
As the question says, I don't think there would be any issues with the user cleaning up the value again, I was looking into the code and on line 154  here https://github.com/onesine/react-tailwindcss-datepicker/blob/master/src/components/Input.tsx#L154 for some reason there is a `!readOnly` making it so you always have to set a new date.
I also wondering why there is not `required` prop?

**Comment by onesine** [comment]
Hi @FerMPY 👋 

Thank you for your feedback.
`readOnly` is normally set to prevent the text field from being modified. You are right, it should be possible to delete it with the button. I had not noticed this behavior.

Thanks for suggesting the `required` option. We will add it very soon.

**Comment by onesine** [comment, marked as answer]
Therefore, we need to :

 - allow the field to be deleted using the button when `readOnly` is true.
 - add the required option

**Reply by FerMPY** [reply]
Awesome thank you for the quick replay :100:

**Reply by alphabetek** [reply]
@onesine any news regarding `required` option?
